File: BubbleSort/bubblesort.py

def sort(array):
    
    for n in range(0,len(array)-1):
        change = False
        for i in range(n,len(array)-1):
            if array[i] > array[i+1]:
                array[i], array[i+1] = array[i+1], array[i]
                change = True
        print(array)
        if change == False: # job done
            return

A = [2,0,5,1,6,3]

sort(A)
 

File: BubbleSort/BubbleSort.scala
object BubbleSort {

    val arr = Array(1,4,5,11,7,3,19,12)

    def main (args: Array[String]) {

        for (a <- 0 to (arr.length - 1)) {
            
            var change = 0
            for (b <- a to (arr.length - 1)) {
                if (arr(b) < arr(a)) {
                    change = 1
                    var aa = arr(a)
                    arr(a) = arr(b)
                    arr(b) = aa
                } 
            }

        }



        for (a <- arr) println(a)

    }

}